feature/experimental/bgp/ate_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go

```diff
@@ -0,0 +1,449 @@
+	nbrPath := statePath.Neighbor(nbr.peerIP)
+	glblPath := statePath.Global()
+
+	// Check BGP peerAS from telemetry.
+	peerAS := gnmi.Get(t, dut, nbrPath.State()).GetPeerAs()
+	if peerAS != nbr.peerAS {
+		t.Errorf("BGP peerAs: got %v, want %v", peerAS, nbr.peerAS)
+	}
+
+	// Check BGP localAS from telemetry.
+	localAS := gnmi.Get(t, dut, nbrPath.State()).GetLocalAs()
+	if localAS != nbr.localAS {
+		t.Errorf("BGP localAS: got %v, want %v", localAS, nbr.localAS)
+	}
+
+	// Check BGP globalAS from telemetry.
+	globalAS := gnmi.Get(t, dut, glblPath.State()).GetAs()
+	if globalAS != nbr.localAS {
+		t.Errorf("BGP globalAS: got %v, want %v", globalAS, nbr.localAS)
+	}
+
+	verifyPrefixesTelemetry(t, dut, 3, nbr.isV4)
+}
+
+func configureATE(t *testing.T, ateParams *bgpNbr, connectionType string, prefixes []string) *ondatra.ATETopology {
+	t.Helper()
+	t.Log("Configure ATE interface")
+	ate := ondatra.ATE(t, "ate")
+	port1 := ate.Port(t, "port1")
+	topo := ate.Topology().New()
+
+	iDut1 := topo.AddInterface(ateSrc.Name).WithPort(port1)
+	iDut1.IPv4().WithAddress(ateSrc.IPv4CIDR()).WithDefaultGateway(dutSrc.IPv4)
+	iDut1.IPv6().WithAddress(ateSrc.IPv6CIDR()).WithDefaultGateway(dutSrc.IPv6)
+
+	bgpDut1 := iDut1.BGP()
+	peer := bgpDut1.AddPeer().WithPeerAddress(ateParams.peerIP).WithLocalASN(ateParams.localAS)
+
+	if connectionType == connInternal {
+		peer.WithTypeInternal()
+	} else {
+		peer.WithTypeExternal()
+	}
+
+	network1 := iDut1.AddNetwork("bgpNeti1")
+	network2 := iDut1.AddNetwork("bgpNeti2")
+	network3 := iDut1.AddNetwork("bgpNeti3")
+
+	if ateParams.isV4 {
+		network1.IPv4().WithAddress(prefixes[0]).WithCount(1)
+		network1.BGP().WithNextHopAddress(ateSrc.IPv4).AddASPathSegment(55000, 4400, 3300)
+
+		network2.IPv4().WithAddress(prefixes[1]).WithCount(1)
+		network2.BGP().WithNextHopAddress(ateSrc.IPv4).AddASPathSegment(55000, 7700)
+		network2.BGP().Communities().WithPrivateCommunities("200:1")
+
+		network3.IPv4().WithAddress(prefixes[2]).WithCount(1)
+		network3.BGP().WithNextHopAddress(ateSrc.IPv4)
+	} else {
+		network1.IPv6().WithAddress(prefixes[0]).WithCount(1)
+		network1.BGP().WithNextHopAddress(ateSrc.IPv6).AddASPathSegment(55000, 4400, 3300)
+
+		network2.IPv6().WithAddress(prefixes[1]).WithCount(1)
+		network2.BGP().WithNextHopAddress(ateSrc.IPv6).AddASPathSegment(55000, 7700)
+		network2.BGP().Communities().WithPrivateCommunities("200:1")
+
+		network3.IPv6().WithAddress(prefixes[2]).WithCount(1)
+		network3.BGP().WithNextHopAddress(ateSrc.IPv6)
+	}
+	return topo
+}
+
+func applyBgpPolicy(policyName string, dut *ondatra.DUTDevice) *oc.NetworkInstance_Protocol {
+	d := &oc.Root{}
+	ni1 := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	niProto := ni1.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	bgp := niProto.GetOrCreateBgp()
+
+	pg := bgp.GetOrCreatePeerGroup("ATE")
+	pg.PeerGroupName = ygot.String("ATE")
+
+	pg.GetOrCreateApplyPolicy().ImportPolicy = []string{policyName}
```

Comment:
BGP RPL should be under Peer-group AFI, could you use RoutePolicyUnderAFIUnsupported to add RPL under PG directly ?

Reply:
@s-nandu to address the code change as requested.

Reply:
Updated with latest commit